src/briefcase/console.py

```diff
@@ -310,76 +319,41 @@ def wait_bar(
         context. Once the new context is complete, the previous Wait Bar message
         is restored.
 
-        :param message: text explaining what is being awaited
+        :param message: text explaining what is being awaited; should end in '...'
         :param done_message: text appended to the message after exiting
         :param transient: if True, remove bar and message from screen after exiting;
             if False (default), the message will remain on the screen without pulsing bar.
         :param markup: whether to interpret Rich styling markup in the message; if True,
             the message must already be escaped; defaults False.
         """
-        # Create the Wait Bar since it is not already active
-        if self._active_wait_bar is None:
-            wait_bar = Progress(
-                TextColumn("    "),
-                BarColumn(bar_width=20, style="black", pulse_style="white"),
-                TextColumn("{task.fields[message]}"),
-                transient=True,
-                console=self.print.console,
-            )
-            # setting start=False causes the progress bar to pulse
-            wait_bar.add_task("", start=False, message=message)
-            try:
-                self.is_output_controlled = True
-                with wait_bar as current_wait_bar:
-                    self._active_wait_bar = current_wait_bar
-                    yield from self._start_wait_bar(
-                        message=message,
-                        done_message=done_message,
-                        transient=transient,
-                        markup=markup,
-                    )
-            finally:
-                self.is_output_controlled = False
-                self._active_wait_bar = None
-
-        # If the Wait Bar is already active, update its message in the console
-        else:
-            wait_bar_task = self._active_wait_bar.tasks[0]
-            orig_message = wait_bar_task.fields["message"]
-            self._active_wait_bar.update(wait_bar_task.id, message=message)
-            try:
-                yield from self._start_wait_bar(
-                    message=message,
-                    done_message=done_message,
-                    transient=transient,
-                    markup=markup,
-                )
-            finally:
-                # Restore Wait Bar message to previous value
-                self._active_wait_bar.update(wait_bar_task.id, message=orig_message)
-
-    def _start_wait_bar(self, message, done_message, transient, markup):
-        """Activate the Wait Bar and manage its exit conditions."""
+        self.is_output_controlled = True
+        wait_bar_task = self._wait_bar.tasks[0]
+        previous_message = wait_bar_task.fields["message"]
+        self._wait_bar.update(wait_bar_task.id, message=message)
         try:
+            self._wait_bar.start()
             yield
         except BaseException:
```

Comment:
I unrolled Rich's `Progress` context manager to manually `start` and `stop` the Bar since `stop` would be called when the first Wait Bar context exited and the Wait Bar would be removed prematurely from the console.

`start()` is reentrant since it simply short-circuits if the Bar is already started.